src/rules/no-nested-type-without-ref.ts

```diff
@@ -0,0 +1,35 @@
+import {
+    Rule,
+    ValidationError,
+    ApiSpec,
+    getAllObjectProperties,
+    hasProperty,
+    getProperty, ApiItem
+} from '../Rule';
+
+export default class NoNestedTypeWithoutRef implements Rule {
+    describe() {
+        return 'When nesting types, the nested type must either be a primitive type or a reference type';
+    }
+
+    validate(json: ApiSpec) {
+        getAllObjectProperties(json, 'components.schemas').forEach(({ object }) => {
+            if (hasProperty(object, 'properties')) {
+               getAllObjectProperties(object, 'properties').forEach(({path, name, object}) => {
+                    if (hasProperty(object, 'type')) {
+                        const type = getProperty(object, 'type');
+                        if (type === 'array' && hasProperty(object, 'items')) {
+                            const items = getProperty(object, 'items');
+                            if (hasProperty(items as ApiItem, 'type') && getProperty(items as ApiItem, 'type') === 'object') {
```

Comment:
If you check if `items` is defined, you don't need to cast to `ApiItem`

Reply:
Yo falso don't need to call `hasProperty()` first, `getProperty()` will return `undefined` if the property doesn't exist, and that will not `=== `object'`